platform/persistence/platform-persistence-commands/src/main/java/org/codice/ddf/persistence/commands/StoreListCommand.java

```diff
@@ -40,49 +40,40 @@
         "User ID to search for notifications. If an id is not provided, then all of the notifications for all users are displayed.",
     multiValued = false
   )
-  private String user;
+  String user;
 
-  private Set<String> headerSet = new TreeSet<String>();
+  private Set<String> headerSet = new TreeSet<>();
 
   @Override
   public void storeCommand() throws PersistenceException {
 
-    List<Map<String, Object>> storeResults;
+    cql = addUserConstraintToCql(user, cql);
 
-    cql = createCql(user, cql);
+    // output the entries
+    // populates the header with the keys from the first entry
+    Function<List<Map<String, Object>>, Integer> listFunction =
+        results -> {
+          int count = 0;
+          for (int i = 0; i < results.size(); i++) {
+            Map<String, Object> curStore = PersistentItem.stripSuffixes(results.get(i));
+            console.println("Result {" + i + "}:");
+            if (headerSet.isEmpty()) {
+              // populates the header with the keys from the first entry
+              headerSet.addAll(curStore.keySet());
+            }
 
-    if (StringUtils.isNotBlank(cql)) {
-      storeResults = persistentStore.get(type, cql);
-    } else {
-      storeResults = persistentStore.get(type);
-    }
-    console.println("Results found: " + storeResults.size() + "\n");
+            for (String curKey : headerSet) {
+              console.println(curKey + ":");
+              console.println("\t" + curStore.get(curKey).toString());
+              count++;
+            }
+          }
+          return count;
+        };
 
-    // output the entries
-    for (int i = 0; i < storeResults.size(); i++) {
-      Map<String, Object> curStore = PersistentItem.stripSuffixes(storeResults.get(i));
-      console.println("Result {" + i + "}:");
-      if (headerSet.isEmpty()) {
-        // populates the header with the keys from the first entry
-        headerSet.addAll(curStore.keySet());
-      }
+    long totalCount = getResults(listFunction);
```

Comment:
❓ Couldn't this command print +100k lines if there were a lot of items or large items in the persistent store?

Should we have a default limit of lines to print like 10 but a command option could increase it?